cEP-0031.md

```diff
@@ -0,0 +1,254 @@
+which the bear will check their existence.
+
+```py
+class FileExistBear(LocalBear):
+    def run(self, required_files: tuple,):
+        for file in required_files:
+            if not os.path.is_file():
+                message = 'File {} not found.'.format(file)
+                yield Result.from_values(origin=self, message=message,)
+```
+
+### 3. RegexLintBear
+
+Issue: [coala/coala-bears#1532][1532]
+
+The task of the bear is to check the regex in strings. This can be done using
+the AnnotationBear to detect all the regex strings and then check for the valid
+regex through the prepared algorithm for each type.
+
+For Python, the bear can be written as follows:
+
+```py
+def is_regex(code, language):
+    """
+    Returns true if the string is a regex.
+    """
+
+class RegexLintBear(LocalBear):
+    def run(self, file, filename, language,):
+        strings = run_local_bear(AnnotationBear)
+        for string in strings:
+            if is_regex(string, language):
+                try:
+                    re.compile(string)
+                except re.error as Exception:
+                    message = 'This is not a valid regex.'
+                    yield Result.from_values(origin=self, message=message)
+```
+
+### 4. FileModeBear
+
+Issue: [coala/coala-bears#2370][2370]
+
+The bear will check the permissions on the files provided by the user
+and to ensure that the file permissions are the one that is expected.
+
+The bear will be used as follows:
+
+```
+[all.mode]
+bears = FileModeBear
+filemode = rw
+
+[all.shell.mode]
+bears = FileModeBear
+filemode = rwx
+```
+
+The bear will first find the permissions of the files specified by the user and
+then if the permissions are not suitable, the bear will try to change those
+permissions into the expected ones. If the bear doesn't have enough permissions
+to do so then the bear will let the user know about this.
+
+```py
+class FileModeBear(LocalBear):
+    def run(self, file, filename, filemode,):
+        stats = os.stat(filename)
+        stats_read = bool(stats.st_mode & stat.S_IRUSR)
+        stats_write = bool(stats.st_mode & stat.S_IWUSR)
+
+        mode = {'r': stats_read, 'w': stats_write}
+        ch_mode = {'r': stat.S_IRUSR, 'w': stat.S_IWUSR}
+
+        # Similarly, by bitwise operations on the st_mode, the relevant
+        # information can be found out.
+
+        change_mode = 0
+        for char in filemode:
+            if not mode[char]:
+                change_mode += ch_mode[char]
+        if not change_mode:
+            os.chmod(change_mode)
+```
+
+### 5. RequirementsCheckBear
+
+Issue: [coala/coala-bears#1113][1113]
+
+The bear will be focused on Python only since they are most prone for
+conflicting requirements.
+
+The implementation is based on the recursive check for the requirements of each
+package in requirement.txt file of the project. The requirements are checked
+through PyPI's JSON API.
```

Comment:
again, dependency management, as above.

The following is quite useless.  The hard part is all of the `get_*` which you havent spec'd out at all.  For a Python only bear, I suggest that you look for the solutions in @dephell rather than in the coala dependency management library.

Reply:
Also this is supposed to be Generic Bear Quality - im not sure how a PIP specific bear fits into that project scope.